src/ucp/proto/proto_am.inl

```diff
@@ -508,13 +508,12 @@ ucp_proto_get_zcopy_threshold(const ucp_request_t *req,
 }
 
 static UCS_F_ALWAYS_INLINE ssize_t
-ucp_proto_get_short_max(const ucp_request_t *req,
-                        const ucp_ep_msg_config_t *msg_config)
+ucp_proto_get_short_max(const ucp_request_t *req, ssize_t max_short)
 {
     return (!UCP_DT_IS_CONTIG(req->send.datatype) ||
             (req->flags & UCP_REQUEST_FLAG_SYNC) ||
-            (!UCP_MEM_IS_HOST(req->send.mem_type))) ?
-           -1 : msg_config->max_short;
+            (!UCP_MEM_IS_ACCESSIBLE_FROM_CPU(req->send.mem_type))) ?
```

Comment:
for AM, we don't need to check req->flags & UCP_REQUEST_FLAG_SYNC..

Reply:
ok, can do another func